File: src/class6/herancas/tarefa/tarefa01/java/Funcionario.java
package class6.herancas.tarefa.tarefa01.java;

public class Funcionario extends Pessoa {
    private float salario;

    public Funcionario(String nome, Data nascimento, float salario) {
        super(nome, nascimento);
        this.salario = salario;
    }

    @Override
    public void imprimeDados() {
        super.imprimeDados();
        System.out.println("Salario: " + salario);
    }

    public float calculaImposto() {
        float taxaImposto = 0.03f;
        float imposto = taxaImposto * this.salario;
        System.out.println("Imposto = R$ " + imposto);
        return imposto;
    }

    public float getSalario() {
        return salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }
}

File: src/exercise04/tarefa/tarefa01/kotlin/Funcionario.kt
package exercise04.tarefa.tarefa01.kotlin

open class Funcionario(nome: String, nascimento: Data, var salario: Float) : Pessoa(nome, nascimento) {

    override fun imprimeDados() {
        super.imprimeDados()
        println("Salario: $salario")
    }

    open fun calculaImposto(): Float {
        val taxaImposto = 0.03f
        val imposto = taxaImposto * salario
        println("Imposto = R$ $imposto")
        return imposto
    }

}